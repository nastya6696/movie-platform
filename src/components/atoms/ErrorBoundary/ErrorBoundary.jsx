import styles from './styles.module.scss';

export const ErrorBoundary = ({children, title}) => {
  const OopsMovieCard = () => (
    <div className={styles.errorCard}>
      <h2>Oops something went wrong!</h2>
      <p>Movie will apear later!</p>
    </div>
  )

  return <>{title ? children : <OopsMovieCard />}</>
}
