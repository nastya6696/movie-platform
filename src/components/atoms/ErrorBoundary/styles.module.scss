@import "src/assets/scss/common";

.errorCard {
  width: 323px;
  height: 455px;
  background-color: $grey;
  border-radius: 4px;
  padding: 100px 30px;
  text-align: center;

  h2 {
    @include fontSizes(30px, 40px);
    font-weight: 400;
  }

  p {
    margin-top: 20px;
    @include fontSizes(16px, 24px);
  }
}
