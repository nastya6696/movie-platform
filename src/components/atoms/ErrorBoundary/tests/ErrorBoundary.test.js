import {create} from "react-test-renderer";
import {ErrorBoundary} from "../ErrorBoundary";

describe('ErrorBoundary', () => {
  it('should create simple ErrorBoundary component wrapper', () => {
    const component = create(
      <ErrorBoundary />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render correct component without ErrorBoundary component wrapper', () => {
    const component = create(
      <ErrorBoundary title='title'>
        <div>Inner content</div>
      </ErrorBoundary>
    );

    expect(component.toJSON()).toMatchSnapshot();
  })
})
