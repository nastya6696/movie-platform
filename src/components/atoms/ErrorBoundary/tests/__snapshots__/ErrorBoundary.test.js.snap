// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should create simple ErrorBoundary component wrapper 1`] = `
<div
  className="errorCard"
>
  <h2>
    Oops something went wrong!
  </h2>
  <p>
    Movie will apear later!
  </p>
</div>
`;

exports[`ErrorBoundary should render correct component without ErrorBoundary component wrapper 1`] = `
<div>
  Inner content
</div>
`;
